lib/lib.es2015.generator.d.ts

```diff
@@ -18,19 +18,19 @@ and limitations under the License.
 /// <reference no-default-lib="true"/>
 
 
-interface Generator extends Iterator<any> { }
+interface Generator<T> extends IterableIterator<T> { }
```

Comment:
What about requiring all fields of the `Iterator` object?

```suggestion
interface Generator<T> extends Required<IterableIterator<T>> { }
```

The `return` and `throw` methods should not be optional as defined in the [ECMAScript 2015 Language Specification](http://www.ecma-international.org/ecma-262/6.0/#sec-generator-objects). This would close #21527.